rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/queue/RedisCircularBuffer.java

```diff
@@ -0,0 +1,188 @@
+package com.lithium.dbi.rdbi.recipes.queue;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.lithium.dbi.rdbi.Handle;
+import com.lithium.dbi.rdbi.RDBI;
+import com.lithium.dbi.rdbi.recipes.cache.SerializationHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Queue;
+
+/**
+ * A circular buffer in redis. Head of the queue is the element at index 0 in redis.
+ */
+public class RedisCircularBuffer<ValueType> implements Queue<ValueType> {
```

Comment:
consider also that your use of this inside our product is going to probably need a way to peek the whole list in redis.